src/Aspire.Hosting.Azure.Sql/AzureSqlServerResource.cs

```diff
@@ -8,10 +8,32 @@ namespace Aspire.Hosting.Azure;
 /// <summary>
 /// Represents an Azure Sql Server resource.
 /// </summary>
-/// <param name="innerResource">The <see cref="SqlServerServerResource"/> that this resource wraps.</param>
-/// <param name="configureConstruct">Callback to populate the construct with Azure resources.</param>
-public class AzureSqlServerResource(SqlServerServerResource innerResource, Action<ResourceModuleConstruct> configureConstruct) : AzureConstructResource(innerResource.Name, configureConstruct), IResourceWithConnectionString
+public class AzureSqlServerResource : AzureConstructResource, IResourceWithConnectionString
```

Comment:
Do we need ISqlServerResource here for composability.

Reply:
That's [Add interfaces to PostgreSQL, Redis, and SqlServer Hosting Resources (dotnet/aspire#6055)](https://github.com/dotnet/aspire/issues/6055)